PhotonSieveDevelopment.js

```diff
@@ -24,92 +24,96 @@ class Sieve {
-       this.pinholeArea = pinholeArea
-       this.pointX = radius * Math.cos(azimuth); 
-       this.pointY = radius * Math.sin(azimuth);       
+       this.pinholeArea = Math.PI * radius ** 2
+       this.point = new Point(parentRingRadius * Math.cos(azimuth), parentRingRadius * Math.sin(azimuth)); 
     }
 }
 
-function distanceBetween(x1, y1, x2, y2) {
-    return Math.hypot(x2 - x1, y2 - y1);
-}
-
-// conflict checking function can safely ignore pinholes placed on rings other than current 
-function getConflicts(conflictRadius, conflictAzimuth) {
-    let conflictArray = [];
-    let candidateX = conflictRadius * Math.cos(conflictAzimuth);
-    let candidateY = conflictRadius * Math.sin(conflictAzimuth);
-    
-    // loop through all pinholes on current ring, check if min spacing constraint is violated
-
-    for(i = 0; i < currentPinholeArray.length; i ++){
-        let conflictDistanceBetweenCenters = distanceBetween(candidateX, currentPinholeArray[i][1], candidateY, currentPinholeArray[i][2]);
-        let conflictDistanceBetweenPerimeters = conflictDistanceBetweenCenters - currentPinholeArray[i][0] - conflictRadius;
-        if(conflictDistanceBetweenPerimeters <= SPACE_MIN){
-            conflictArray.push(i);
-        }
-    return conflictArray.length
-    }
+function distanceBetween(a, b) {
+    return Math.hypot(b.x - a.x, b.y - a.y);
 }
 
 function randMinMax(min, max) {
     return Math.random() * ((max - min) + min);
- }
-
-let currentPinholeArray = []; 
+ } 
  
 //create new pinhole sieve object
 const photonSieve = new Sieve(0, 0, 0);
 
 //loop through fresnelZonePlate, add a Ring for each even-numbered zone
-for(i = 1; i < fresnelZonePlate.length; i = i + 2){
-    let zone = i - 1;
+for(let i = 1; i < fresnelZonePlate.length; i = i + 2){
     let zoneWidth = fresnelZonePlate[i] - fresnelZonePlate[i-1];
     let centerlineRadius = fresnelZonePlate[i-1] + zoneWidth / 2;
     let ringArea = (Math.PI * fresnelZonePlate[i] ** 2) - (Math.PI * fresnelZonePlate[i-1] ** 2);
-    var ring = new Ring(i-1, centerlineRadius, zoneWidth, 0, 0);
+    var ring = new Ring(i-1, centerlineRadius, zoneWidth);
 
     //add pinholes to ring until total area constraint is fulfilled
     while(ring.openArea < AREA_PROP * ringArea){
         let minRadius = Math.max(DIA_MIN, zoneWidth * .25);
         let candidateRadius = randMinMax(minRadius, zoneWidth * MAX_FACTOR);
-        let candidateAzimuth = Math.random() * Math.PI * 2;
-        if(getConflicts(candidateAzimuth, candidateRadius) = 0){
-            let pinhole = new Pinhole(candidateRadius, candidateAzimuth, 0);         
-            currentPinholeArray.push([pinhole.radius, pinhole.pointX, pinhole.pointY]);
-            pinhole.pinholeArea = Math.PI * radius ** 2;
-            ring.openArea = ring.openArea + pinhole.pinholeArea;
-            ring.numOfPinholes = ring.numOfPinholes + 1;
-            }
+        let candidateAzimuth = Math.random() * Math.PI * 2 * ring.centerlineRadius;  

```

Comment:
REVIEW THIS: I assume you meant to have the centerlineRadius here as part of formula